File: pyt/isSymmetrical.py
"""

Create a function that takes a number as an argument and returns True or False
depending on whether the number is symmetrical or not.

A number is symmetrical when it is the same as its reverse.

"""


def is_symmetrical(num):
    n = len(str(num))
    if n == 1:
        return True
    else:
        if str(num)[::-1] == str(num):
            return True
    return False

File: c#/IsSymmetrical.cs
        public static bool IsSymmetrical(int num) 

        /* 
        Create a function that takes a number as an argument and returns true or false depending on whether the number is symmetrical or not.
        
        A number is symmetrical when it is the same as its reverse.
        */

        {

            string numToString = num.ToString();
            int half = numToString.Length / 2;
            string firstHalf = numToString.Substring(0, half);
            string secondHalf = numToString.Substring(numToString.Length - half);
            
            char[] secondHalfToCharArray = secondHalf.ToCharArray();
            Array.Reverse(secondHalfToCharArray);
            string reversedSecondHalf = new string(secondHalfToCharArray);


            return firstHalf.Equals(reversedSecondHalf);

        }